pylint/extensions/typing.py

```diff
@@ -0,0 +1,255 @@
+        return
+
+    # We want our line numbers to be 1 indexed, however fromlineno and to lineno
+    # are 0 indexed so add 1, except fromlineno always starts one line after the
+    # def or class so just don't add 1 to that one.
+    for line in range(node.fromlineno, node.tolineno + 1):
+        scope_map[line] = node
+
+    for child in node.get_children():
+        fill_scope_map(scope_map, child)
+
+def inject_imports(module):
+    # type: (NodeNG) -> NodeNG
+    assert isinstance(module, astroid.Module)
+
+    if not module.pure_python:
+        return module
+
+    stream = module.stream()
+    if not stream:
+        return module
+
+    # Don't bother checking files without annotations.
+    encoding = module.file_encoding or 'utf-8'
+    if not re.search(r'#\s*type:', stream.read().decode(encoding)):
+        return module
+
+    # Build a scope_map for faster lookups.
+    scope_map = [module] * (module.tolineno + 1)
+    fill_scope_map(scope_map, module)
+
+    # Tokenize the file looking for those juicy # type: annotations.
+    stream.seek(0)
+    module.injected_lines = []
+    tokens = tokenize.generate_tokens(lambda: stream.readline().decode(encoding))
+    for tok_type, tok_val, (lineno, _), _, _ in tokens:
+        if tok_type == tokenize.COMMENT:
+            expr = expr_for_comment(tok_val)
+            if expr:
+                # Now convert that expression to an Assign node and insert it in the module.
+                assign = build_assign(expr, lineno)
+                scope = scope_map[lineno]
+                i = find_insert_point(scope, lineno)
+                if isinstance(scope.body[i], astroid.Pass) and len(scope.body) == 1:
+                    # If what we have is a single pass statement then replace it with the assign.
+                    # This is to avoid 'unnecessary-pass' warnings.
+                    scope.body[i] = assign
+                else:
+                    scope.body.insert(i, assign)
+                assign.parent = scope
+                module.injected_lines.append(lineno)
+
+    return module
+
+
+class IgnoreMyPyWarnings(BaseChecker):
+    """Checker that disables certain events on the added lines.
+
+    Note: Creating a checker was the only way I found to do this.
+
+    'multiple-statements': The way we add expressions to assignments will cause multiple
+    statements on the same line so that's the only way to fix it.
+
+    """
+    __implements__ = IRawChecker
+
+    name = 'ignore-mypy'
+    msgs = {
+        # Apparently you have to define something so that this runs.
+        'W9999': ('Unused',
+                  'ignore-mypy-errors',
+                  'Used to disable mypy errors on added lines'),
+    }
+
+    def process_module(self, module):
+        # type: (NodeNG) -> None
+        for line in getattr(module, 'injected_lines', []):
+            self.linter.disable('multiple-statements', scope='module', line=line)
```

Comment:
This is pretty gross.  Another solution to this is to somehow mark the Nodes that are type annotations somehow and have the multiple-statements checker skip those nodes explicitly.